app/controllers/katello/api/v2/content_view_puppet_modules_controller.rb

```diff
@@ -24,13 +23,8 @@ def index
     param :id, String, :desc => N_("the id of the puppet module to associate")
     param :uuid, String, :desc => N_("the uuid of the puppet module to associate")
     def create
-      params[:content_view_puppet_module][:uuid] = @puppet_module.uuid if @puppet_module
-
-      @puppet_module = ContentViewPuppetModule.create!(puppet_module_params) do |puppet_module|
-        puppet_module.content_view = @view
-      end
-
-      respond :resource => @puppet_module
+      params[:content_view_puppet_module][:uuid] ||= PuppetModule.find_by(id: params[:id]).try(:uuid) if params[:id]
```

Comment:
If I use an id which doesn't exist in the db, I get this error message which is misleading:

```
$ hammer content-view puppet-module add --id 42 --content-view-id 3
Could not add the puppet module:
  Validation failed: Invalid puppet module parameters specified.  Either 'uuid' or 'name' and 'author' must be specified.
```

If you change this to:

```ruby
params[:content_view_puppet_module][:uuid] ||= PuppetModule.find(params[:id]).uuid if params[:id]
```

This will raise an error if the puppet module is not found, you should be good.